webrender/src/render_backend.rs

```diff
@@ -490,10 +497,24 @@ impl RenderBackend {
         // implementations - a single flush for each webgl
         // context at the start of a render frame should
         // incur minimal cost.
-        for (_, webgl_context) in &self.webgl_contexts {
-            webgl_context.make_current();
-            webgl_context.apply_command(WebGLCommand::Flush);
-            webgl_context.unbind();
+        // glFlush is not enough in some GPUs.
+        // glFlush doesn't guarantee the completion of the GL commands when the shared texture is sampled.
+        // This leads to some graphic glitches on some demos or even nothing being rendered at all (GPU Mali-T880).
+        // glFinish guarantees the completion of the commands but it may hurt performance a lot.
+        // Sync objects are the recommended way to ensure that textures are ready in OpenGL 3.0+.
+        // They are more performant than glFinish and guarantee the completion of the GL commands.
+        for (id, webgl_context) in &self.webgl_contexts {
+            if self.dirty_webgl_contexts.contains(&id) {
+                webgl_context.make_current();
+                // Call FenceSync and ClientWaitSync to ensure that textures are ready.
+                let sync = webgl_context.gl().fence_sync(gl::SYNC_GPU_COMMANDS_COMPLETE, 0);
```

Comment:
Do you mean that they should be called using WebGLCommand enum? (as it was done with the existing glFlush call)